supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiReferenceFilesMapping.scala

```diff
@@ -71,57 +71,58 @@ protected trait PipelinesApiReferenceFilesMappingOperations {
     }
   }
 
-  private def getReferenceFileToValidatedGcsPathMap(referenceFiles: Set[ReferenceFile]): IO[Map[ReferenceFile, ValidFullGcsPath]] = {
+  private def getReferenceFileToValidatedGcsPathMap(referenceFiles: Set[ReferenceFile]): Map[ReferenceFile, ValidFullGcsPath] = {
     val filesAndValidatedPaths = referenceFiles.map {
       referenceFile => (referenceFile, GcsPathBuilder.validateGcsPath(s"gs://${referenceFile.path}"))
     }.toMap
 
-    val (filesWithValidPaths, filesWithInvalidPaths) = filesAndValidatedPaths.partition(_._2.isInstanceOf[ValidFullGcsPath])
+    val filesWithValidPaths = filesAndValidatedPaths.collect {
+      case (referenceFile, validPath: ValidFullGcsPath) => (referenceFile, validPath)
+    }
+    val filesWithInvalidPaths = filesAndValidatedPaths.collect {
+      case (referenceFile, invalidPath: InvalidFullGcsPath) => (referenceFile, invalidPath)
+    }
 
     if (filesWithInvalidPaths.nonEmpty) {
-      IO.raiseError(new InvalidGcsPathsInManifestFileException(filesWithInvalidPaths.keySet.map(_.path).toList))
+      throw new InvalidGcsPathsInManifestFileException(filesWithInvalidPaths.keySet.map(_.path).toList)
```

Comment:
I'm not sure whether this change was caused by me saying I would prefer function inputs to be simple types instead of `IO`s. If so, sorry, I didn't mean to imply that throwing exceptions is better than wrapping these unsafe operations in `IO`s. I think we should almost never throw exceptions directly and almost always prefer `IO`s, especially in `trait` functions like these.

All I meant was that we should do the `flatMap`s above functions rather than within them, so that each conversion is written as a function of a simple type, and we can compose them together if we need to. Eg instead of 
```
def functionA(t1: Type1): IO[Type2] = ...
def functionB(t2: IO[Type2]): IO[Type3] = ...

def functionC(t1: Type1): IO[Type3] = functionB(functionA(t1))
```

I would prefer:
```
def functionA(t1: Type1): IO[Type2] = ...
def functionB(t2: Type2): IO[Type3] = ...

def functionC(t1: Type1): IO[Type3] = for {
  t2 <- functionA(t1)
  t3 <- functionB(t2)
} yield t3
```